### PR TITLE
code refactoring, building training manager class

### PR DESCRIPTION
**Pull request by xXKeNdAmAmAsTeRXx**
move training steps to class methods as we use them in two different scripts

### CODE REVIEW

**Review by Ajver — CHANGES_REQUESTED** [review]

File: continue.py
[the remark this answers is not in the document]
Comment (xXKeNdAmAmAsTeRXx) [reply continue.py:1]: # Separating one epoch
One idea is that we can store one epoch as a function of training manager. Then treat  training_step as private function called inside epoch function. Otherwise we can prepare external perform epoch function. In this approaches the continue and train scripts only differs with start and epoch loop.

[the remark this answers is not in the document]
Comment (xXKeNdAmAmAsTeRXx) [reply continue.py:1]: ## TrainingManager
We should keep training manger as it sorts out code and it is ease to reuse. On the other hand, I am not a big fun keeping logging system inside TrainingManager class as for me it is different thing.

[the remark this answers is not in the document]
Comment (xXKeNdAmAmAsTeRXx) [reply continue.py:1]: ## creating_epoch function
Also we could make a higher order function that returns function performing one epoch along with set loaders and other external components. That allow as to keep all training logic in training manager class.

File: TrainingManager.py [hunk 1]
```diff
@@ -48,7 +42,7 @@ def __init__(self):
         
 
     
-    def training_step(self, train_loader):
+    def training_step(self, train_loader) -> float:
```
Comment (Ajver) [inline TrainingManager.py:45]: I meant the `train_loader` argument should be statically typed:

`def training_step(self, train_loader: DataLoader) -> float:`

File: TrainingManager.py [hunk 2]
```diff
@@ -66,7 +60,7 @@ def training_step(self, train_loader):
         return train_loss / len(train_loader)
     
 
-    def validation_step(self, val_loader):
+    def validation_step(self, val_loader) -> tuple[float]:
```
Comment (Ajver) [inline TrainingManager.py:63]: The `val_loader` argument should be statically typed:

`def validation_step(self, val_loader: DataLoader) -> tuple[float]:`

Also: the returned type `tuple[float]` is wrong. This function actually returns:

`def validation_step(self, val_loader: DataLoader) -> (np.ndarray, np.ndarray, float):`

BUT: why do we even return the arrays of images?? I think it's better to calculate all the scores in this function and just return these (i.e. return auc_score, accuracy, loss... etc)

**Review by Ajver — CHANGES_REQUESTED** [review]

File: TrainingManager.py (hunk 2 above)
Comment (xXKeNdAmAmAsTeRXx) [reply TrainingManager.py:63]: What do you mean by array of images? The function returns predicted classes.

File: continue.py [hunk 3]
```diff
@@ -159,36 +92,25 @@ def main():
...
-            model_path = os.path.join(SAVE_MODEL_PATH, f"Best_Loss.pth")
-            torch.save(checkpoint, model_path)
+            train.save_checkpoint("Best_Loss", epoch, best_val_loss, best_auc, logs)
             print(f"Best loss model saved")
 
         if val_auc > best_auc:
             best_auc = val_auc
-            model_path = os.path.join(SAVE_MODEL_PATH, f"Best_AUC.pth")
-            torch.save(checkpoint, model_path)
+            train.save_checkpoint("Best_AUC", epoch, best_val_loss, best_auc, logs)
```
[the remark this answers is not in the document]
Comment (xXKeNdAmAmAsTeRXx) [reply continue.py:103]: I would not store them in trainer class as we use them only for model saving reference.

**Review by Ajver — CHANGES_REQUESTED** [review]

File: train.py [hunk 4]
```diff
@@ -12,176 +12,30 @@
...
-            for imgs, labels in val_loader:
-                imgs, labels = imgs.to(device), labels.to(device)
-                outputs = model(imgs)
-                loss = criterion(outputs, labels)
-                val_loss += loss.item()
-                probs = torch.softmax(outputs, dim=1)
-                val_preds.append(probs.cpu().numpy())
-                val_labels.extend(labels.cpu().numpy())
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--continue_training', action="store", default=None)
```
Comment (Ajver) [inline train.py:28]: If continue flag requires a checkpoint path, maybe rename it to:
`load_checkpoint` (and then, I'd NOT allow the short `-c` arg)
